### PR TITLE
feat: Header validation with exact header names & header count match validation Updated

### PR DESCRIPTION
**Pull request by choudsc**
JIRA link (if applicable)

https://tools.hmcts.net/jira/browse/RDCC-1327

Change description

feat: Header validation with exact header names & header count match validation.

Does this PR introduce a breaking change? (check one with "x")
No

### CODE REVIEW

File: src/integrationTest/resources/application-leaf-integration.yml [hunk 1]
```diff
@@ -25,6 +26,7 @@ leaf-route:
     processor-class: JudicialContractTypeProcessor
     mapper-class: JudicialContractTypeRowMapper
     csv-binder-object: JudicialContractType
+    route-header: ContractID,ContractDescription
```
Comment (abhijitdiwan) [inline src/integrationTest/resources/application-leaf-integration.yml:29]: please update leaf headers. Take headers from Sourav.for role there 3 headrs.Simliraly for other leaf

File: src/test/java/uk/gov/hmcts/reform/juddata/camel/helper/JrdTestSupport.java [hunk 2]
```diff
@@ -15,8 +26,15 @@
 import uk.gov.hmcts.reform.juddata.camel.binder.JudicialUserRoleType;
 import uk.gov.hmcts.reform.juddata.camel.route.beans.RouteProperties;
 
+@TestPropertySource(properties = {"spring.config.location=classpath:application-integration.yml"})
 public class JrdTestSupport {
 
+    @Value("${route.judicial-user-profile-orchestration.route-header}")
+    public static String fileHeader;
+
+    public static Map readYmlAsMap = readYmlAsMap("test-header.yaml");
```
Comment (abhijitdiwan) [inline src/test/java/uk/gov/hmcts/reform/juddata/camel/helper/JrdTestSupport.java:35]: do you need to maintain separate yaml?

File: src/main/java/uk/gov/hmcts/reform/juddata/camel/processor/HeaderValidationProcessor.java [hunk 3]
```diff
@@ -53,25 +50,20 @@
...
-            if (!java.lang.reflect.Modifier.isStatic(field.getModifiers())) {
-                csvFields.add(field);
-            }
-        }
+        List<String> routHeader = Arrays.asList(routeProperties.getRouteHeader().split(","));
+        List<String> fileHeader = Arrays.asList(header);
 
         //Auditing in database if headers are missing
-        if (header.length > csvFields.size()) {
+        if (!routHeader.equals(fileHeader)) {
```
Comment (abhijitdiwan) [inline src/main/java/uk/gov/hmcts/reform/juddata/camel/processor/HeaderValidationProcessor.java:66]: use notEquals. avoid !.

File: src/test/java/uk/gov/hmcts/reform/juddata/camel/helper/JrdTestSupport.java (hunk 2 above)
Comment (choudsc) [reply src/test/java/uk/gov/hmcts/reform/juddata/camel/helper/JrdTestSupport.java:35]: its for testing purpose only and  hardcoding header will looks messy

**Pull request closed without merging**